lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2308,7 +2623,8 @@ class PrimalGenCloner final
     mapValue(sei, originalDirectResult);
     // Checkpoint the pullback.
     SILValue pullback = vjpDirectResults.back();
-    getPrimalInfo().addPullbackDecl(sei, pullback->getType());
+    // TODO: Check whether it's necessary to reabstract getter pullbacks.
+    getPrimalInfo().addPullbackDecl(sei, getOpType(pullback->getType()));
```

Comment:
It is totally necessary. You can write a computed/stored property that returns a generic parameter in the parent type to test it. Write a custom VJP to avoid triggering `struct_element_addr` differentiation.

Reply:
I tried to write tests that would trigger this code path (requiring getter pullback reabstraction) but failed. Initial experiments:

```swift
protocol Addable : Differentiable & FloatingPoint {
  @differentiable(wrt: (x, y))
  static func add(_ x: Self, _ y: Self) -> Self
}

struct Foo<Scalar : Differentiable> : Differentiable
  // where Scalar : AdditiveArithmetic, Scalar.TangentVector : AdditiveArithmetic, Scalar.CotangentVector : AdditiveArithmetic
  where Scalar : Addable, Scalar.TangentVector : AdditiveArithmetic, Scalar.CotangentVector : AdditiveArithmetic
{
  var member: Scalar

  var computed: Scalar {
    // return .zero + member
    return Scalar.add(.zero, member)
  }
}

func test<Scalar : Differentiable>(_ x: Foo<Scalar>) {
  print(pullback(at: Foo<Scalar>(member: Scalar.zero), in: { foo in foo.computed }))
}
```

I think we can implement getter pullback reabstraction lazily, once we have an effective test.